Userland/DevTools/HackStudio/ProjectTemplate.cpp

```diff
@@ -96,24 +93,17 @@ Result<void, ByteString> ProjectTemplate::create_project(ByteString const& name,
         // Generate a namespace-safe project name (replace hyphens with underscores)
         auto namespace_safe = name.replace("-"sv, "_"sv, ReplaceMode::All);
 
-        pid_t child_pid;
         char const* argv[] = { postcreate_script_path.characters(), name.characters(), path.characters(), namespace_safe.characters(), nullptr };
 
-        if ((errno = posix_spawn(&child_pid, postcreate_script_path.characters(), nullptr, nullptr, const_cast<char**>(argv), environ))) {
-            perror("posix_spawn");
-            return ByteString("Failed to spawn project post-create script.");
-        }
+        pid_t child_pid = TRY(Core::System::posix_spawn(postcreate_script_path, nullptr, nullptr, const_cast<char**>(argv), environ));
 
         // Command spawned, wait for exit.
-        int status;
-        if (waitpid(child_pid, &status, 0) < 0)
-            return ByteString("Failed to spawn project post-create script.");
-
-        int child_error = WEXITSTATUS(status);
+        auto waitpid_result = TRY(Core::System::waitpid(child_pid, 0));
+        int child_error = WEXITSTATUS(waitpid_result.status);
```

Comment:
pre-existing, but if `WIFSIGNALED(waitpid_result.status)`, then this will give garbage. 

This isn't quite the correct way to spawn a child and wait for it to exit. iirc @DanShaders added a new common-ish set of APIs for that (or extended the ones we already had). Food for next PR.

Reply:
:yakplus:

Reply:
Done in https://github.com/SerenityOS/serenity/pull/22805 - that API is much nicer!